### PR TITLE
Add `ignore_index` keyword arg in `dropna` and `drop_duplicates` (Part of GH624)

### PR DESCRIPTION
**Pull request by hyperc54**
- [x] Should cover two bullet points of #624 (dropna and drop_duplicates for both Series and DFs) 
- [x] Tests added

- Additionally add overloads to dataframe `drop_duplicates` method
- Additionally convert existing tests of DF `drop_na` method to use `assert_type`.

### CODE REVIEW

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -900,15 +902,35 @@ class DataFrame(NDFrame, OpsMixin):
...
+    @overload
+    def drop_duplicates(
+        self,
+        subset: Hashable | Iterable[Hashable] | None = ...,
+        *,
+        keep: NaPosition | _bool = ...,
+        inplace: Literal[True] = ...,
+        ignore_index: _bool = ...,
+    ) -> None: ...
+    @overload
```
Comment (Dr-Irv) [inline pandas-stubs/core/frame.pyi:925]: Reverse these two overloads.  Pattern should look like what is in `fillna()`.  First you'd have `inplace: Literal[True]` (without the ellipses), then you'd have `Literal[False] = ...`

File: pandas-stubs/core/series.pyi [hunk 2]
```diff
@@ -726,15 +726,27 @@ class Series(IndexOpsMixin[S1], NDFrame):
...
     ) -> Series[S1]: ...
     @overload
     def drop_duplicates(
-        self, *, keep: NaPosition | Literal[False] = ..., inplace: Literal[True]
+        self,
+        *,
+        keep: NaPosition | Literal[False] = ...,
+        inplace: Literal[True],
+        ignore_index: _bool = ...,
     ) -> None: ...
```
Comment (Dr-Irv) [inline pandas-stubs/core/series.pyi:742]: Same comment as within `frame.pyi` - reverse the 2 overloads (looks like they were not using the consistent pattern before)

**Comment by hyperc54** [comment]
Right thanks! definitely missed the subtlety. I applied the suggestions.

**Review by Dr-Irv — APPROVED** [review]
thanks @hyperc54

**Pull request merged**